exporter/clickhouselogsexporter/exporter.go

```diff
@@ -321,18 +393,56 @@ func (e *clickhouseLogsExporter) pushToClickhouse(ctx context.Context, ld plog.L
 				}
 			}
 		}
-		dbWriteStart := time.Now()
-		err = statement.Send()
-		stats.RecordWithTags(ctx,
-			[]tag.Mutator{
-				tag.Upsert(exporterKey, component.DataTypeLogs.String()),
-				tag.Upsert(tableKey, DISTRIBUTED_LOGS_TABLE),
-			},
-			writeLatencyMillis.M(int64(time.Since(dbWriteStart).Milliseconds())),
+
+		insertResourcesStmtV2, err = e.db.PrepareBatch(
+			ctx,
+			fmt.Sprintf("INSERT into %s.%s", databaseName, DISTRIBUTED_LOGS_RESOURCE_V2),
+			driver.WithReleaseConnection(),
 		)
 		if err != nil {
-			return fmt.Errorf("StatementSend:%w", err)
+			return fmt.Errorf("couldn't PrepareBatch for inserting resource fingerprints :%w", err)
+		}
+
+		for bucketTs, resources := range resourcesSeen {
+			for resourceLabels, fingerprint := range resources {
+				insertResourcesStmtV2.Append(
+					resourceLabels,
+					fingerprint,
+					bucketTs,
+				)
+			}
+		}
+
+		var wg sync.WaitGroup
+		chErr := make(chan error, 3)
+		chDuration := make(chan time.Duration, 3)
+
+		wg.Add(3)
+		go send(statement, chDuration, chErr, &wg)
+		go send(insertLogsStmtV2, chDuration, chErr, &wg)
+		go send(insertResourcesStmtV2, chDuration, chErr, &wg)
+		wg.Wait()
+		close(chErr)
+
+		// store the duration to send the data
+		for _, table := range []string{DISTRIBUTED_LOGS_TABLE, DISTRIBUTED_LOGS_TABLE_V2, DISTRIBUTED_LOGS_RESOURCE_V2} {
+			duration := <-chDuration
+			stats.RecordWithTags(ctx,
+				[]tag.Mutator{
+					tag.Upsert(exporterKey, component.DataTypeLogs.String()),
+					tag.Upsert(tableKey, table),
+				},
+				writeLatencyMillis.M(int64(duration.Milliseconds())),
+			)
+		}
```

Comment:
The order of the durations written to `chDuration` won't necessarily be the same as the order of the tables defined in iteration.